libs/SalesforceSDK/src/com/salesforce/androidsdk/config/BootConfig.java

```diff
@@ -77,9 +83,24 @@
 	private boolean shouldAuthenticate;
 	private boolean attemptOfflineLoad;
 	private String pushNotificationClientId;
+	private String unauthenticatedStartPage;
 
 	private static BootConfig INSTANCE = null;
 
+	BootConfig(JSONObject config, boolean isHybrid) {
```

Comment:
I added some package-visible constructors, for unit test support.

Reply:
Same question as above. Do we need this?

Reply:
Do you mean the hybrid flag specifically, or the constructors generally?  Without the constructors, BootConfig instantiation is hard-wired into reading from www/bootconfig.json. Given that the BootConfig only exposes read-only values from that process, it's an onerous approach to formulate isolated tests that tweak values in that configuration to test their validity.

Reply:
Yes - I was referring to the hybrid flag specifically. For example, this constructor makes sense only when `isHybrid` is `true` and the next one makes sense only when it's `false`.

Reply:
Could we have a package level static method that takes in a filename instead and avoid these constructors? We could revert the changes made to the existing static method and simply call the package level static method from there with the default filename. In the tests, you could pass in a bunch of fake boot configs.

Reply:
Or a static method that takes `JSONObject`.

Reply:
> Could we have a package level static method that takes in a filename instead and avoid these constructors?

And then the thinking for the tests is canned bootconfigs in the text context, each representing a different test case?  That could be a good approach, so test consumers could see the configuration in its expected form, so to speak.  Let me know if that's what you're getting at, and I can rework it.

Reply:
Yes - exactly!